server/routes/requests.js

```diff
@@ -322,4 +323,55 @@ router.get('/getConfirmed/:id/:timeOffset', passport.authenticate('jwt', {sessio
     }
 })
 
+const generateNewAndExistingActionItems = async (existingActionItems, newActionItems) => {
+    const existingActionItemsIds = existingActionItems.map(actionItem => actionItem.value).flat()
+    let existingActionItemsRecords = await actionItemSchema.find().where('_id').in(existingActionItemsIds).exec()
+    // create action itmes added
+    if (newActionItems.length) {
+        for (let i = 0; i < newActionItems.length; i++) {
+            // check to see if action items name already exists
+            let actionItemsExists = await actionItemSchema.find({name: newActionItems[i].value})
+            if (!actionItemsExists.length) {
+                var newActionItemCreated = new actionItemSchema({
+                    name: newActionItems[i].value
+                })
+                await newActionItemCreated.save()
+                existingActionItemsRecords.push(newActionItemCreated)
+            } else {
+                // user accidentally added an interest that already exists as a new interest
+                existingActionItemsRecords.push(actionItemsExists[0])
+            }
+        }
+    }
+    return existingActionItemsRecords;
+}
+
+const getUniqueActionItems = (allActionItems) => {
+    let allUniqueNames = new Set()
+    let uniqueActionItems = []
+    for (let i = 0; i < allActionItems.length; i++) {
+        if (!allUniqueNames.has(allActionItems[i].name)) {
+            allUniqueNames.add(allActionItems[i].name)
+            uniqueActionItems.push(allActionItems[i])
+        }
+    }
+    return uniqueActionItems
+}
+
+router.patch('/actionItems/:id/', passport.authenticate('jwt', {session: false}), async (req, res, next) => {
+    try {
+        let requestId = req.body.requestId;
```

Comment:
one you pass the requestId as url params, you can get it here like `req.params.id`